src/traces/scattermapbox/convert.js

```diff
@@ -294,7 +318,7 @@ function makeSymbolGeoJSON(calcTrace, gd) {
 
 function getFillFunc(attr) {
     if(Lib.isArrayOrTypedArray(attr)) {
-        return function(v) { return v; };
+        return function(i) { return attr[i]; };
```

Comment:
This actually seems just fine for symbol and text, since those are both strings / string arrays. But what happens for angles if the user provides an array of numbers-as-strings? A single number as a string will get converted to an actual number by `coerce`, but for performance we don't reach into arrays to do this (until the `calc` step - that would be the one benefit of including `angle` in calc, if it was being used from there here, but I don't recommend that because `scattermapbox` shares calc with `scattergeo` so this would need to be a special case.)

It may be that this is just fine - assuming mapbox casts this angle to a number consistently, and treats non-numbers as 0. Can you confirm that this works?

Reply:
ok so I tested this, and yes basically coerce seems to turn a string into an actual number which means it gets interpreted as an acceptable angle, but if provided as an array, then string values get passed on to mapbox which treats non-numbers as 0.

I'm not sure whether that's an acceptable behavior, and honestly I don't think I know enough about the code to figure out how to make it work the way it *should* without having to have a few back and forths and me diving deeper into other functions. So if you don't mind if this is not acceptable, it'd be great if you could suggest a fix.

Reply:
I think https://github.com/plotly/plotly.js/pull/4575/commits/1e1458b160b41362e7aa1b7963b25b2208405d4b should do it 🙏